### PR TITLE
More platform installer tests

### PR DESCRIPTION
**Pull request by dzuelke**
- covers extension ordering in generated repo explicitly (see also https://github.com/composer/composer/pull/10110)
- test forward compat with packages that are built for our upcoming Composer 2 installer
- test userland packages providing native extensions edge cases

GUS-W-8105408
GUS-W-9889098

### CODE REVIEW

**Comment by dzuelke** [comment]
(tests are currently failing due to a `/dev/std{err,out}` permissions regression in CR with `heroku run`)

File: .circleci/config.yml [hunk 1]
```diff
@@ -23,9 +23,16 @@ jobs:
...
             gem install bundler --version="~> 1"
       - run:
-          name: Install PHP
+          name: Update apt
           command: |
             sudo apt-get update
+      - run:
+          name: Install Python
+          command: |
+            sudo apt-get install python
```
Comment (edmorley) [inline .circleci/config.yml:32]: Since these are single commands, the multi-line syntax can be removed (the `|`). Also I wonder if you might be able to get away with `--no-install-recommends` to speed up CI slightly.

**Review by edmorley — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (dzuelke) [reply .circleci/config.yml:32]: fcd2482 (and no recommended installs, so no benefit in this case)
